stdlib/gc.mli

```diff
@@ -491,39 +495,70 @@ module Memprof :
       ('minor, 'major) tracker ->
-      unit
-    (** Start the sampling with the given parameters. Fails if
-       sampling is already active.
+      t
+
+    (** Start sampling for a profile with the given parameters. Raises
+       an exception if a profile is already sampling in the current
+       domain.
 
        The parameter [sampling_rate] is the sampling rate in samples
        per word (including headers). Usually, with cheap callbacks, a
        rate of 1e-4 has no visible effect on performance, and 1e-3
-       causes the program to run a few percent slower
+       causes the program to run a few percent slower.
 
        The parameter [callstack_size] is the length of the callstack
        recorded at every sample. Its default is [max_int].
 
        The parameter [tracker] determines how to track sampled blocks
        over their lifetime in the minor and major heap.
 
-       Sampling is temporarily disabled when calling a callback
-       for the current thread. So they do not need to be re-entrant if
-       the program is single-threaded. However, if threads are used,
-       it is possible that a context switch occurs during a callback,
-       in this case the callback functions must be re-entrant.
+       Sampling is temporarily disabled on the current thread when
+       calling a callback, so callbacks do not need to be re-entrant
+       if the program is single-threaded and single-domain. However,
+       if threads or multiple domains are used, it is possible that
+       several callbacks will run in parallel. In this case, callback
+       functions must be re-entrant.
+
+       Note that a callback may be postponed slightly after the actual
+       event. The callstack passed to an allocation callback always
+       accurately reflects the allocation, but the program state may
+       have evolved between the allocation and the call to the
+       callback.
+
+       If a new thread or domain is created when profiling is active,
+       the child thread or domain joins that profile (using the same
+       [sampling_rate], [callstack_size], and [tracker] callbacks).
 
-       Note that the callback can be postponed slightly after the
-       actual event. The callstack passed to the callback is always
-       accurate, but the program state may have evolved. *)
+       An allocation callback is generally run by the thread which
+       allocated the block. If the thread exits or the profile is
+       stopped before the callback is called, the callback may be run
+       by a different thread.
+
+       Each callback is generally run by the domain which allocated
+       the block. If the domain terminates or the profile is stopped
+       before the callback is called, the callback may be run by a
+       different domain.
+
+       Different domains may run different profiles simultaneously.
+       *)
 
     val stop : unit -> unit
-    (** Stop the sampling. Fails if sampling is not active.
 
```

Comment:
I think the usual convention in OCaml library (or at least in this file) is to not add an empty line between the val declaration and its documentation.